route/build/src/tracer.f90

```diff
@@ -0,0 +1,364 @@
+         'at ', NETOPO_in(segIndex)%REACHID
+ end if
+
+ call comp_reach_mb(NETOPO_in(segIndex)%REACHID, idxDW, Cupstream, Clat, RCHFLX_out(iens,segIndex), verbose, lakeFlag=.false.)
+
+ END SUBROUTINE constituent_rch
+
+ ! *********************************************************************
+ ! subroutine: solve diffuisve wave equation
+ ! *********************************************************************
+ SUBROUTINE comp_mass_flux(idxRoute,       & ! input: routing method index
+                           rch_param,      & ! input: river parameter data structure
+                           Cupstream,      & ! input: discharge from upstream
+                           Clat,           & ! input: lateral mass flux into chaneel [mg/s]
+                           isHW,           & ! input: is this headwater basin?
+                           subrch_state,   & ! inout: state at a sub-reach
+                           rflux,          & ! inout: reach flux at a reach
+                           verbose,        & ! input: reach index to be examined
+                           ierr,message)
+ ! ----------------------------------------------------------------------------------------
+ ! Description
+ !
+ ! ----------------------------------------------------------------------------------------
+ USE globalData, ONLY : nMolecule   ! number of internal nodes for finite difference (including upstream and downstream boundaries)
+ implicit none
+ ! Argument variables
+ integer(i4b), intent(in)        :: idxRoute       ! routing method index
+ type(RCHPRP), intent(in)        :: rch_param      ! River reach parameter
+ real(dp),     intent(in)        :: Cupstream      ! total discharge at top of the reach being processed
+ real(dp),     intent(in)        :: Clat           ! lateral discharge into chaneel [m3/s]
+ logical(lgt), intent(in)        :: isHW           ! is this headwater basin?
+ type(SUBRCH), intent(inout)     :: subrch_state   ! curent states at sub-reaches
+ type(STRFLX), intent(inout)     :: rflux          ! current reach fluxes
+ logical(lgt), intent(in)        :: verbose        ! reach index to be examined
+ integer(i4b), intent(out)       :: ierr           ! error code
+ character(*), intent(out)       :: message        ! error message
+ ! Local variables
+ real(dp)                        :: max_outCC      ! maximum possible constituent discharge [mg/s]
+ real(dp)                        :: reach_vol      ! water volume [m3]
+ real(dp)                        :: reach_mass     ! water volume [m3]
+ real(dp)                        :: solute_out     ! mass flux out [mg/s]
+ real(dp)                        :: solute_per_vol ! mass concentration in channel [mg/m3]
+ character(len=strLen)           :: cmessage       ! error message from subroutine
+
+ ierr=0; message='comp_mass_flux/'
+
+ rflux%ROUTE(idxRoute)%reach_solute_mass(0) = rflux%ROUTE(idxRoute)%reach_solute_mass(1)
```

Comment:
Does the 0 and 1 refer to the move from current time to previous time?

Reply:
0 and 1 will be replaced with parameters. This change (issue https://github.com/ESCOMP/mizuRoute/issues/532) will be made in another PR.